src/main/java/frc/team2412/robot/Controls.java

```diff
@@ -136,6 +138,25 @@ public Controls(Subsystems s) {
 			// 						this,
 			// 						codriveController.leftBumper()));
 		}
+		if (DRIVEBASE_ENABLED && APRILTAGS_ENABLED) {
+			new Trigger(s.rotateToSpeaker)
+					.whileTrue(
+							s.drivebaseSubsystem
+									.rotateToAngle(
+											() -> {
+												double currentTimestamp = Timer.getFPGATimestamp();
+												double robotToTagAngleTimestamp =
+														s.apriltagsProcessor.getLastRotatedAngleTimestamp();
+												var robotAngle = s.drivebaseSubsystem.getPose().getRotation();
+												var robotToTagAngle = s.apriltagsProcessor.getLastRotatedAngle();
```

Comment:
You might move this after the if() {} below.  No reason to have the computer do the work to get robotToTagAngle if you will never use it when the if is true.